### PR TITLE
doc: add a blog for Notation v1.2.0 release

### PR DESCRIPTION
**Pull request by yizha1**
Updates:
- Create a blog for Notation v1.2.0 release

Signed-off-by: Yi Zha <yizha1@microsoft.com>

### CODE REVIEW

File: content/en/blog/2024/announcing-notation-v1-2.md
Comment (FeynmanZhou) [inline content/en/blog/2024/announcing-notation-v1-2.md:28]: I think it worth mentioning Notation timestamp is built on the [tspclient-go library](https://github.com/notaryproject/tspclient-go) here.

Comment (Two-Hearts) [inline content/en/blog/2024/announcing-notation-v1-2.md:28]: Since `Notary Project specifications` has a section itself, we don't need to repeat it here.
```suggestion
Since this release, Notation supports RFC 3161 compliant Timestamps. Digital signatures must be generated within the certificate's validity period, as expired certificates compromise the signature's trustworthiness. Trusted timestamps extend the trust of signatures created within certificate validity, allowing successful signature verification even after certificates have expired. Learn more at the document [how to sign and verify artifacts in OCI-compliant registries with timestamping](../../docs/user-guides/how-to/timestamps).
```

Comment (FeynmanZhou) [inline content/en/blog/2024/announcing-notation-v1-2.md:28]: According to the [RFC 3161](https://datatracker.ietf.org/doc/html/rfc3161.html) and [Notary Project Spec](https://github.com/notaryproject/specifications/blob/main/specs/signing-and-verification-workflow.md), is it much consistent to use `supports RFC 3161 compliant Timestamp standard (or protocol)` and `Trusted timestamping` in this paragraph? I seldomly see the term `timestamps`.

File: content/en/blog/2024/announcing-notation-v1-2.md
Comment (FeynmanZhou) [inline content/en/blog/2024/announcing-notation-v1-2.md:32]:
```suggestion
Notation CLI now offers the `armv7` binary, enabling its usage in environments with ARM32 processors.

```

File: content/en/blog/2024/announcing-notation-v1-2.md
Comment (FeynmanZhou) [inline content/en/blog/2024/announcing-notation-v1-2.md:36]: Suggest using a relative link `[quick start](../../docs/quickstart)` here. Otherwise, this link will always point to the latest version in the future.

**Review by FeynmanZhou — COMMENTED** [review]
Thanks @yizha1 . Overall looks good to me. Just a few small suggested changes.

File: content/en/blog/2024/announcing-notation-v1-2.md
Comment (Two-Hearts) [inline content/en/blog/2024/announcing-notation-v1-2.md:24]: nit:
```suggestion
In Feb 2024, the Open Container Initiative (OCI) community released version 1.1.0, which includes the [OCI image specification v1.1.0](https://github.com/opencontainers/image-spec/releases/tag/v1.1.0) and the [OCI distribution specification v1.1.0](https://github.com/opencontainers/distribution-spec/releases/tag/v1.1.0). Notation now adheres to the OCI spec v1.1.0, leading to the deprecation of the experimental flag `--allow-referrers-api`. A new flag, `--force-referrers-tag` (default to true), has been introduced to the `notation sign` command, enabling users to choose the referrers tag schema over the [referrers API](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#enabling-the-referrers-api) if the registry they are using does not yet support the referrers API. The `notation verify/list/inspect` commands now attempt to use the referrers API first and automatically fall back to the [referrers tag schema](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#referrers-tag-schema) if the referrers API is not supported by the registry.

```

**Review by Two-Hearts — APPROVED** [review]
LGTM

File: content/en/blog/2024/announcing-notation-v1-2.md
Comment (FeynmanZhou) [inline content/en/blog/2024/announcing-notation-v1-2.md:28]:
```suggestion
Since this release, Notation supports RFC 3161 compliant timestamping. Digital signatures must be generated within the certificate's validity period, as expired certificates compromise the signature's trustworthiness. Timestamping extends the trust of signatures created within certificate validity, allowing successful signature verification even after certificates have expired. Notation's timestamping feature is built on top of the [tspclient-go](https://github.com/notaryproject/tspclient-go) library.

```

Comment (FeynmanZhou) [reply content/en/blog/2024/announcing-notation-v1-2.md:28]: Nit: the hyperlink is not rendered as expected.

**Review by FeynmanZhou — COMMENTED** [review]
A tiny markdown format issue

**Review by FeynmanZhou — APPROVED** [review]
LGTM. This PR can be merged after its referenced doc PR #407 is merged.

**Review by FeynmanZhou — APPROVED** [review]
LGTM

**Review by JeyJeyGao — APPROVED** [review]
LGTM, but IANAM

File: content/en/blog/2024/announcing-notation-v1-2.md
Comment (shizhMSFT) [inline content/en/blog/2024/announcing-notation-v1-2.md:24]: > A new flag, `--force-referrers-tag` (default to true)

You may want to mention that this flag will be default to `false` in future releases.

Comment (shizhMSFT) [reply content/en/blog/2024/announcing-notation-v1-2.md:24]: Examples of golang release notes can be found.

In [golang 1.22](https://go.dev/doc/go1.22) release notes:

> We plan to include an API migration tool in a future release, likely Go 1.23.

> We may change the default value of x509usepolicies in Go 1.23, making Policies the default field for marshaling.

> Go 1.23 will require macOS 11 Big Sur or later.

In [golang 1.23](https://go.dev/doc/go1.23) release notes:

> The [x509sha1 GODEBUG setting](https://go.dev/pkg/crypto/x509#InsecureAlgorithmError) will be removed in the next Go major release (Go 1.24). This will mean that crypto/x509 will no longer support verifying signatures on certificates that use SHA-1 based signature algorithms.

> Go 1.24 will require Linux kernel version 3.17 or later, with an exception that systems running 3.10 or later will continue to be supported if the kernel has been patched to support the getrandom system call.

File: content/en/blog/2024/announcing-notation-v1-2.md
Comment (shizhMSFT) [inline content/en/blog/2024/announcing-notation-v1-2.md:34]: It seems redundant to mention ARM32 processors. Is it for IoT scenarios?

**Review by FeynmanZhou — APPROVED** [review]
LGTM

File: content/en/blog/2024/announcing-notation-v1-2.md
Comment (shizhMSFT) [inline content/en/blog/2024/announcing-notation-v1-2.md:27]: do you want to replace `may` to `will`?

**Review by shizhMSFT — DISMISSED** [review]
LGTM

**Review by FeynmanZhou — DISMISSED** [review]
LGTM

**Review by shizhMSFT — DISMISSED** [review]
LGTM

**Review by vaninrao10 — APPROVED** [review]
LGTM

File: content/en/blog/2024/announcing-notation-v1-2.md
Comment (FeynmanZhou) [inline content/en/blog/2024/announcing-notation-v1-2.md:33]:
```suggestion
Learn more at the document [how to sign and verify artifacts in OCI-compliant registries with timestamping](/docs/user-guides/how-to/timestamping/).

```

Comment (FeynmanZhou) [reply content/en/blog/2024/announcing-notation-v1-2.md:33]: image

**Review by FeynmanZhou — DISMISSED** [review]
LGTM

File: content/en/blog/2024/announcing-notation-v1-2.md
Comment (whalelines) [inline content/en/blog/2024/announcing-notation-v1-2.md:41]:
```suggestion
You can follow the [quick start guide](/docs/quickstart-guides/quickstart-sign-image-artifact/) to try Notation v1.2.0 for basic signing and verification workflow.

```

File: content/en/blog/2024/announcing-notation-v1-2.md
Comment (yizha1) [reply content/en/blog/2024/announcing-notation-v1-2.md:27]: @whalelines if you mean the note, the breaking changes will be done in Notation v2, the current release v1.2.0 does not introduce any breaking changes.

File: content/en/blog/2024/announcing-notation-v1-2.md
Comment (yizha1) [reply content/en/blog/2024/announcing-notation-v1-2.md:24]: If all registries support referrers API, then there is no need for adding this flag `--force-referrers-tag`.  We want referrers API to be the default one, this is the goal. However, not all registries currently support referrers API. Another reason is we do not want to introduce breaking changes in v1.2.0, since version v1.10 and prior versions by default use tag schema. 

In Notation v2.0, we will introduce the breaking change, so that by default Notation cli will use referrers API, however, user can choose to use tag schema even if registries support referrers API by setting `--force-referrers-tag` to `true`

**Review by FeynmanZhou — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

File: content/en/blog/2024/announcing-notation-v1-2.md
Comment (whalelines) [reply content/en/blog/2024/announcing-notation-v1-2.md:24]: Read this sentence from the blog post:

> The `notation verify/list/inspect` commands attempt to use the referrers API first and automatically fall back to the [referrers tag schema](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#referrers-tag-schema) if the referrers API is not supported by the registry.

This is the proper OCI 1.1 client behavior, i.e., the client must check to see if the registry supports the referrers API before trying to use it and if the registry does not support the referrers API the client must use the referrers tag fallback.

You say:

> If all registries support referrers API, then there is no need for adding this flag `--force-referrers-tag`.

If notation is OCI 1.1 compliant, it does not matter if all registries support the referrers API or not. The _only_ use of the `--force-referrers-tag` command-line option is if the client wants to use the referrers tag fallback _even if a registry supports the referrers API._ If a registry does not support the referrers API, the client _has_ to use the referrers tag fallback, there is no other option. You acknowledge this in this later statement:

> user can choose to use tag schema even if registries support referrers API by setting `--force-referrers-tag` to `true`

Therefore, this statement in the blog is misleading/wrong:

> A new flag, `--force-referrers-tag` (default to `true`), has been introduced to the `notation sign` command, enabling users to choose the referrers tag schema over the [referrers API](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#enabling-the-referrers-api) if the registry they are using does not yet support the referrers API.

The end of the sentence should read "… even if the registry they are using supports the referrers API."

File: content/en/blog/2024/announcing-notation-v1-2.md
Comment (whalelines) [reply content/en/blog/2024/announcing-notation-v1-2.md:27]: I was referring to the removal of the `--allow-referrers-api` command-line option, but it appears that may not have been removed in this release.

Comment (yizha1) [reply content/en/blog/2024/announcing-notation-v1-2.md:27]: @whalelines `--allow-referrers-api` is an experimental flag, so removing it is not a breaking change. This change was based on the discussion here https://github.com/notaryproject/notation/issues/892

File: content/en/blog/2024/announcing-notation-v1-2.md
Comment (yizha1) [reply content/en/blog/2024/announcing-notation-v1-2.md:24]: @whalelines I think I understood your point now. You mean with the current default behavior, tag schema is used over the referrers API even if the registry they are using supports the referrers API. I will create a PR to amend the blog. Thanks.

**Pull request merged**
